src/Form/Search.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace Drupal\islandora_collection_search\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\Core\Cache\CacheableMetadata;
+
+use Drupal\islandora\Controller\DefaultController as IslandoraController;
+use Drupal\islandora_solr\IslandoraSolrQueryProcessor;
+
+/**
+ * Form for searching within a given collection (or site wide).
+ */
+class Search extends FormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'islandora_collection_search_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form_state->loadInclude('islandora_collection_search', 'inc', 'includes/search.form');
+    $form_state->loadInclude('islandora_collection_search', 'inc', 'includes/utilities');
+
+    $cache = CacheableMetadata::createFromRenderArray($form);
+    $cache->addCacheTags([IslandoraController::LISTING_TAG]);
+    $cache->addCacheableDependency($this->config('islandora_collection_search.settings'));
+
+    $form['simple'] = [
+      '#type' => 'container',
+      '#attributes' => [
+        'class' => [
+          'container-inline',
+        ],
+      ],
+    ];
+
+    $collection = islandora_collection_search_get_collection_pid();
+    $options = [
+      'all' => $this->t('All collections'),
+    ];
+    $default_search = 'all';
+    if ($collection) {
+      // See if the current object has children, and if so make it available for
+      // searching otherwise get its parent.
+      $qp = new IslandoraSolrQueryProcessor();
+      $cache->addCacheableDependency($qp);
+      $qp->buildQuery('*:*');
+      $qp->solrParams['fq'][] = strtr(
+        '!is_memberofcollection:"info:fedora/!pid" OR !is_member:"info:fedora/!pid"',
+        [
+          '!is_memberofcollection' => $this->config('islandora_solr.settings')->get('islandora_solr_member_of_collection_field'),
+          '!is_memberof' => $this->config('islandora_solr.settings')->get('islandora_solr_member_of_field'),
+          '!pid' => $collection,
+        ]
+      );
+
+      $qp->executeQuery(FALSE);
+      // If no children need to find the immediate parent to search for instead.
+      if (!$qp->islandoraSolrResult['response']['numFound']) {
+        module_load_include('inc', 'islandora_basic_collection', 'includes/utilities');
+        $child_object = islandora_object_load($collection);
+        $parents = islandora_basic_collection_get_parent_pids($child_object);
+        $collections = array_filter(array_combine($parents, 'islandora_object_load', $parents));
```

Comment:
This... seems in the wrong order... or... what is this supposed to do?... `array_combine()` doesn't take 3 args, in any case:
```
Warning: array_combine() expects exactly 2 parameters, 3 given in Drupal\islandora_collection_search\Form\Search->buildForm() (line 71 of modules/contrib/islandora_collection_search/src/Form/Search.php). 
```

Reply:
For future reference, the [`drupal_map_assoc()` call](https://github.com/discoverygarden/islandora_collection_search/pull/23/files#diff-da7ed641f7e362bfd3c0b948ee67f0dfL53) would be more something like:
```php
$collections = array_combine($parents, array_map('islandora_object_load', $parents));
```
... though really, the keys of the `$collections` array are never used... probably an artifact of older implementation, before we did the other label things.